packages/rs-drive-abci/src/query/v0/mod.rs

```diff
@@ -115,13 +115,24 @@ impl<C> Platform<C> {
             "/identity" => {
                 let GetIdentityRequest { id, prove } =
                     check_validation_result_with_data!(GetIdentityRequest::decode(query_data));
-                let identity_id: Identifier = check_validation_result_with_data!(id.try_into());
+                let identity_id: Identifier = check_validation_result_with_data!(id
+                    .try_into()
+                    .map_err(|_| QueryError::InvalidArgument(
+                        "id must be a valid identifier (32 bytes long)".to_string()
+                    )));
                 let response_data = if prove {
                     let proof = check_validation_result_with_data!(self.drive.prove_full_identity(
                         identity_id.into_buffer(),
                         None,
                         &platform_version.drive
                     ));
+
+                    if proof.is_empty() {
```

Comment:
I don't think the proof ever be empty. In case of non existence it will be proof of non existence? @QuantumExplorer @lklimek right?

Reply:
Agree, we should expect proof of non-existence in this case.

But this check is correct here, empty proof is an error condition and we should return error to the client.

Reply:
Fixed